### ISSUE TITLE
withQueryParam with String value is considered as a Seq[Char]

### ISSUE THREAD

**Issue by julien-truffaut** [http4s/http4s#187]
```scala
Uri().withQueryParam("foo", "value with space").renderString == "/?foo=v&foo=a&foo=l&foo=u&foo=e&foo= &foo=w&foo=i&foo=t&foo=h&foo= &foo=s&foo=p&foo=a&foo=c&foo=e"
```

also, shall we url encode the query parameters?

```scala
Uri().withQueryParam("foo", Seq("value with space")).renderString == "/?foo=value with space"
```

**Comment by bryce-anderson** [comment]
First one works for me. 
```scala
scala> Uri().withQueryParam("foo", "value with space").renderString
res0: String = /?foo=value with space

scala> Uri().withQueryParam("foo", "value").renderString
res1: String = /?foo=value
```
Do you have any implicits in scope that could be causing this? I think we should remove the ```implicit val charQueryParamEncoder```; if someone wants a single char they can do a single char String and that should remove this possibility.

I agree, they should probably be rendered as encoded Strings.

**Comment by julien-truffaut** [comment]
@bryce-anderson I have an implicit conversion between String and Seq[Char] but I don't know where does it come from. I thought it was from Prelude

**Comment by bryce-anderson** [comment]
We should remove the ```charQueryParamEncoder``` to make sure its not happening again. I'll do that in a PR.

**Comment by bryce-anderson** [comment]
This was closed via #189 and #190
